pyproject.toml

```diff
@@ -22,29 +22,43 @@ classifiers = [
 ]
 
 [tool.poetry.dependencies]
+# default required dependencies
 python = "^3.8.1"
 pydantic = "^1.10.10"
-requests = "^2.31.0"
-tiktoken = "0.4.0"
-beautifulsoup4 = "^4.12.2"
 openai = "^0.27.8"
 langchain = ">=0.0.271,<0.1.0"
-pandas = ">=1.0.5"
 pygments = "^2.15.1"
-google-api-python-client = "^2.90.0"
-chispa = "^0.9.2"
-grpcio = ">=1.56.0"
-plotly = "^5.15.0"
-pyarrow = ">=4.0.0"
-grpcio-status = ">=1.56.0"
-faiss-cpu = "^1.7.4"
-sentence-transformers = "^2.2.2"
-torch = ">=2.0.0, !=2.0.1"
-
-[tool.poetry.dev-dependencies]
+
+# optional dependencies, opted into by groups
+
+# dependencies for create
+requests = { version = "^2.31.0", optional = true }
+tiktoken = { version = "0.4.0", optional = true }
+beautifulsoup4 = { version = "^4.12.2", optional = true }
+google-api-python-client = { version = "^2.90.0", optional = true }
+
+# dependencies for plot
+pandas = { version = ">=1.0.5", optional = true, extras = ["all"]}
+grpcio = { version = ">=1.56.0", optional = true}
```

Comment:
this one is for spark connect. We can have a group for spark connect.

Reply:
we might need a way to detect whether a sparksession is a spark connect session. But we can have it as follow-up.